src/main/java/fr/skytasul/guardianbeam/Laser.java

```diff
@@ -391,15 +391,23 @@ protected void sendDestroyPackets(Player p) throws ReflectiveOperationException
 		@Override
 		public void moveStart(Location location) throws ReflectiveOperationException {
 			this.start = location;
-			if (main != null) moveFakeEntity(start, guardianID, guardian);
-		}
+			if (main != null) {
+        		initGuardian();
+        		if (guardian == null) {
+            		for (Player p : show) {
+            		   Packets.sendPackets(p, createGuardianPacket, metadataPacketGuardian);
+            		}
+        	} else {
+            	moveFakeEntity(start, guardianID, guardian);
+        	}
+    }
 		
 		@Override
 		public void moveEnd(Location location) throws ReflectiveOperationException {
 			this.end = location;
 			if (main != null) {
+        		initSquid();
```

Comment:
Same here, this could be optimized.

Reply:
The packets are "re-created" every time so they use the new end location, would this work with just updating the createSquidPacket?